v2/internal/testcommon/vcr/redact.go

```diff
@@ -16,6 +16,22 @@ import (
 	"github.com/Azure/azure-service-operator/v2/internal/testcommon/creds"
 )
 
+type Redactor struct {
```

Comment:
Creating an explicit `Redactor` to capture all the desired redactions is a good idea. It makes the coupling visible, which is much less fragile.